lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -72,50 +72,95 @@ void SetMessageManagerHandler::SetAmfSessionContext(
   // Print the message from AMF
   PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request_cpy));
   response_callback(Status::OK, SmContextVoid());
-  /*The Event Based main_thread invocation and runs to handle session state*/
-  m5g_enforcer_->get_event_base().runInEventBaseThread(
-      [this, response_callback, request_cpy]() {
-        // extract values from proto
-        auto imsi = request_cpy.common_context().sid().id();
-        std::string dnn =
-            request_cpy.common_context().apn();  // may not required for demo-1
-        std::string session_id = id_gen_.gen_session_id(imsi);
-
-        MLOG(MDEBUG) << "Requested session from UE with IMSI: " << imsi
-                     << " Generated sessioncontext ID" << session_id;
-        /*reach complete message from proto message*/
-        SessionConfig cfg = m5g_build_session_config(request_cpy);
-
-        /*check if it's initial message*/
-        if ((cfg.rat_specific_context.m5gsm_session_context().rquest_type() ==
-             INITIAL_REQUEST) &&
-            (cfg.common_context.sm_session_state() == CREATING_0)) {
-          /* it's new UE establisment request and need to create the session
-           * context
-           */
-          MLOG(MDEBUG)
-              << "AMF request type INITIAL_REQUEST and session state CREATING";
-          /* Read the SessionMap from global session_store,
-           * if it is not found, it will be added w.r.t imsi
-           */
+

```

Comment:
isn't this function changed in this pr? https://github.com/magma/magma/pull/3366/files

Maybe you need to rebase